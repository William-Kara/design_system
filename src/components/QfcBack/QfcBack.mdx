import { ArgTypes, Meta } from '@storybook/blocks';
import QfcBack from '@/components/QfcBack/QfcBack.vue';

<Meta title="Elements/Back" />

# Definition

QfcBack is used to go back to the previous page.

## Props

The following props are available for the QfcBack component:

<ArgTypes of={QfcBack} />

# Usage

To use QfcBack, style needs to be imported separately.

```sass
@import '@website/design-system-vuejs/dist/styles/components/QfcBanner.scss';
```

```html
<template>
  <QfcBack label="Lorem" href="#" />
</template>
```
