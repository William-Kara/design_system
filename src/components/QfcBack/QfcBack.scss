@import '../../styles/bulma/layout/hero';

.qfc-back {
  display:table;
  color: $white;
  &:hover {
    color: $white;
  }
  &__icon  {
    font-size: 28px;
    padding: 1rem;
  }
  &__link  {
    font-size: $fs-m;
    display:table-cell;
    vertical-align:middle;
  }
}


