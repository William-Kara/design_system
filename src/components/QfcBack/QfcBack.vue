<template>
  <a class="qfc-back" :href="href">
    <span class="qfc-back__icon material-icons-outlined" width="168" height="18">arrow_back</span>
    <span class="qfc-back__link">{{ label }}</span>
  </a>
</template>

<script lang="ts" setup>
interface Props {
  label: string;
  href: string;
}
withDefaults(defineProps<Props>(), {
  label: 'Retour à l’accueil',
  href: '/',
});
</script>
<style lang="scss" scoped src="./QfcBack.scss" />
