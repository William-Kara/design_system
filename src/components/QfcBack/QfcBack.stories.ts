import QfcBack from './QfcBack.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof QfcBack> = {
  component: QfcBack,
  title: 'Elements/Back',
  argTypes: {},
};
export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {},
};
