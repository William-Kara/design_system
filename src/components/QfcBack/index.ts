import QfcBack from './QfcBack.vue';

export { QfcBack };
export default QfcBack;
