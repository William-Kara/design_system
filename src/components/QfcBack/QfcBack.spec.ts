import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcBack from './QfcBack.vue';

let wrapper: any = null;

const propsValues = {
  label: 'Retour à l’accueil',
  href: '/',
};

beforeEach(() => {
  wrapper = shallowMount(QfcBack, {
    propsData: propsValues,
  });
});

describe('QfcBack Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
