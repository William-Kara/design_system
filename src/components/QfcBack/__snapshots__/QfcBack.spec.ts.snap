// Vitest Snapshot v1

exports[`QfcBack Component > should render correctly 1`] = `"<a data-v-b842fc57=\\"\\" class=\\"qfc-back\\" href=\\"/\\"><span data-v-b842fc57=\\"\\" class=\\"qfc-back__icon material-icons-outlined\\" width=\\"168\\" height=\\"18\\">arrow_back</span><span data-v-b842fc57=\\"\\" class=\\"qfc-back__link\\">Retour à l’accueil</span></a>"`;
